event.go

```
@@ -166,23 +171,39 @@ func CancelEventHandle(handle uint64) error {
 	return nil
 }
 
-/* Get the first event in the log, for testing */
-func getTestEventHandle() (EventHandle, error) {
-	wideQuery, _ := syscall.UTF16PtrFromString("*")
-	wideChannel, _ := syscall.UTF16PtrFromString("Application")
+type QueryResult struct {
+	handle syscall.Handle
+}
+
+func QueryChannel(channel, query string) (*QueryResult, error) {
```

check params for valid values?

---

What kind of invalid values?  Empty strings?  The API function should return an appropriate error in that case.